### PR TITLE
feat: add Windows support to run_rust

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add threaded Windows handling to run_rust
- cover Windows cargo streaming with tests

## Testing
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_68aba5eb8ac08322a55189f5f875ed32

## Summary by Sourcery

Implement Windows-specific handling in run_rust._run_cargo to stream cargo stdout and stderr via threads, include a pre-check for missing streams, and maintain the selector-based approach for non-Windows platforms; update test setup and add Windows-focused tests for streaming success, non-zero exit codes, and missing stream scenarios.

New Features:
- Enable streaming of cargo stdout and stderr on Windows using threading in run_rust._run_cargo

Enhancements:
- Add a pre-check in run_rust._run_cargo to raise an error if stdout or stderr streams are not captured

Tests:
- Add tests for Windows cargo streaming success, non-zero exit code handling, and missing stdout/stderr errors
- Clear cached coverage_parsers module in test setup to avoid import issues

### CODE REVIEW

File: .github/actions/generate-coverage/tests/test_scripts.py [hunk 1]
```diff
@@ -163,6 +164,36 @@ def test_run_rust_success(tmp_path: Path, shell_stubs: StubManager) -> None:
...
+        "run_rust_win", script_dir / "run_rust.py"
+    )
+    mod = importlib.util.module_from_spec(spec)
+    monkeypatch.setitem(sys.modules, "run_rust_win", mod)
+    monkeypatch.setattr(os, "name", "nt")
+    assert spec.loader is not None
+    spec.loader.exec_module(mod)
+
+    def fake_echo(line: str, *, err: bool = False, nl: bool = True) -> None:
+        print(line, end="" if not nl else "\n", file=sys.stderr if err else sys.stdout)
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/tests/test_scripts.py:187]: Please provide a prompt for an AI agent

**Comment by leynos** [comment]
@coderabbitai review

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 2]
```diff
@@ -111,23 +114,52 @@ def _run_cargo(args: list[str]) -> str:
...
+    if os.name == "nt":
+
+        def pump(src: t.TextIO, *, to_stdout: bool) -> None:
+            for line in iter(src.readline, ""):
+                if to_stdout:
+                    typer.echo(line, nl=False)
+                    stdout_lines.append(line.rstrip("\n"))
+                else:
+                    typer.echo(line, err=True, nl=False)
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

File: .github/actions/generate-coverage/tests/test_scripts.py [hunk 3]
```diff
@@ -163,6 +164,36 @@ def test_run_rust_success(tmp_path: Path, shell_stubs: StubManager) -> None:
...
+def test_run_cargo_windows(
+    shell_stubs: StubManager,
+    monkeypatch: pytest.MonkeyPatch,
+    capsys: pytest.CaptureFixture[str],
+) -> None:
+    """``_run_cargo`` streams output correctly on Windows."""
+    script_dir = Path(__file__).resolve().parents[1] / "scripts"
+    monkeypatch.setenv("PATH", shell_stubs.env["PATH"])
+    shell_stubs.register("cargo", stdout="out-line\n", stderr="err-line\n")
+    monkeypatch.syspath_prepend(script_dir)
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/tests/test_scripts.py:177]: Has this now been resolved?

File: .github/actions/generate-coverage/tests/test_scripts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/tests/test_scripts.py]: Has this now been resolved?

**Comment by leynos** [comment]
@sourcery-ai review

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 4]
```diff
@@ -110,24 +114,56 @@ def _run_cargo(args: list[str]) -> str:
...
+    if os.name == "nt":
+        thread_exceptions: list[Exception] = []
+
+        def pump(src: t.TextIO, *, to_stdout: bool) -> None:
+            try:
+                for line in iter(src.readline, ""):
+                    if to_stdout:
+                        typer.echo(line, nl=False)
+                        stdout_lines.append(line.rstrip("\n"))
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?
